src/validators/mod.rs

```diff
@@ -317,124 +318,123 @@ fn build_specific_validator<'a, T: BuildValidator>(
-        // integers
-        int::IntValidator,
-        // boolean
-        bool::BoolValidator,
-        // floats
-        float::FloatValidator,
-        // tuples
-        tuple::TupleBuilder,
-        // list/arrays
-        list::ListValidator,
-        // sets - unique lists
-        set::SetValidator,
-        // dicts/objects (recursive)
-        dict::DictValidator,
-        // None/null
-        none::NoneValidator,
-        // functions - before, after, plain & wrap
-        function::FunctionBuilder,
-        // function call - validation around a function call
-        call::CallValidator,
-        // literals
-        literal::LiteralBuilder,
-        // any
-        any::AnyValidator,
-        // bytes
-        bytes::BytesValidator,
-        // dates
-        date::DateValidator,
-        // times
-        time::TimeValidator,
-        // datetimes
-        datetime::DateTimeValidator,
-        // frozensets
-        frozenset::FrozenSetValidator,
-        // timedelta
-        timedelta::TimeDeltaValidator,
-        // introspection types
-        is_instance::IsInstanceValidator,
-        is_subclass::IsSubclassValidator,
-        callable::CallableValidator,
-        // arguments
-        arguments::ArgumentsValidator,
-        // default value
-        with_default::WithDefaultValidator,
-        // chain validators
-        chain::ChainValidator,
-        // lax or strict
-        lax_or_strict::LaxOrStrictValidator,
-        // generator validators
-        generator::GeneratorValidator,
-        // custom error
-        custom_error::CustomErrorValidator,
-        // json data
-        json::JsonValidator,
-        // url types
-        url::UrlValidator,
-        url::MultiHostUrlValidator,
-        // recursive (self-referencing) models
-        definitions::DefinitionRefValidator,
-        definitions::DefinitionsBuilder,
-    )
+    let schema: &PyDict = schema.downcast()?;
+    let type_: &str = schema.get_as_req(intern!(schema.py(), "type"))?;
+    let val = match type_ {
+        // The left hand side of this match is a 1:1 match with the `type` field we use as a discriminator
+        // in our CoreSchema union
+        // The right hand side is _generally_ a 1:1 match but there are cases where we use a `Builder`
+        // on the right that may have internal logic to return different validators or just build a
+        // a more complex validator (e.g. building a union of isinstance validators or something).
+        // So to get from Python -> Rust implementation you should trace the `type` on the left hand side
+        // of the match and then if the right hand side is a `{Type}Validator` you've found the implementation.
+        // If the right hand side is a `{Type}Builder` you'll have to look into it's `build()` method to see
+        // what it actually returns.
+        // TODO: sort alphabetically? By left hand side string or right hand side type? Or RHS type's module filename?
```

Comment:
Not sure i agree about alphabetical, in theory validators are ordered logically in `core_schema.py`, then the order here should ideally match that.

If we make a change, we should make it everywhere.

Reply:
agreed, it should be consistent.

alphabetic is obviously simplest. problem is it puts closes related validators like `model` and `typed-dict` or `int` and `float` far apart.